micro_dl/inference/image_inference.py

```diff
@@ -745,10 +745,17 @@ def predict_2d(self, chan_slice_meta):
             target_stack.append(cur_target.astype(np.float32))
         pred_stack = np.concatenate(pred_stack, axis=0) #zcyx
         target_stack = np.concatenate(target_stack, axis=0)
+
         # Stack images and transpose (metrics assumes cyxz format)
         if self.image_format == 'zyx':
-            pred_stack = np.transpose(pred_stack, [1, 2, 3, 0])
-            target_stack = np.transpose(target_stack, [1, 2, 3, 0])
+            if len(np.shape(pred_stack)) == 4:
+                pred_stack = np.transpose(pred_stack, [1, 2, 3, 0])
+                target_stack = np.transpose(target_stack, [1, 2, 3, 0])
+            else:
+                pred_stack = pred_stack[:, :, 0, :, :]
+                pred_stack = np.transpose(pred_stack, [1, 2, 3, 0])
+                target_stack = target_stack[:, :, 0, :, :]
+                target_stack = np.transpose(target_stack, [1, 2, 3, 0])
```

Comment:
You can use `self.input_depth` for `if` statement to handle the extra dimension in the output of 2.5D model as here:
https://github.com/czbiohub/microDL/blob/2eff0f015d4877daeeb3ad87670836e8495fba5c/micro_dl/inference/image_inference.py#L556

The logic is easier to understand than using the dimension of the array although both would work. Also you don't need to repeat
`pred_stack = np.transpose(pred_stack, [1, 2, 3, 0])`
  `target_stack = np.transpose(target_stack, [1, 2, 3, 0])`